devicetypes/smartthings/zwave-door-temp-sensor.src/zwave-door-temp-sensor.groovy

```diff
@@ -0,0 +1,311 @@
+                attributeState("default", label:'${currentValue}°',
+                        backgroundColors: [
+                                [value: 31, color: "#153591"],
+                                [value: 44, color: "#1e9cbb"],
+                                [value: 59, color: "#90d2a7"],
+                                [value: 74, color: "#44b621"],
+                                [value: 84, color: "#f1d801"],
+                                [value: 95, color: "#d04e00"],
+                                [value: 96, color: "#bc2323"]
+                        ]
+                )
+            }
+        }
+        valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+            state "battery", label: '${currentValue}% battery', unit: ""
+        }
+
+        main "contact"
+        details(["contact", "battery"])
+    }
+}
+
+private getCommandClassVersions() {
+    [
+            0x20: 1, // Basic
+            0x25: 1, // Switch Binary
+            0x30: 1, // Sensor Binary
+            0x31: 5, // Sensor Multilevel
+            0x80: 1, // Battery
+            0x84: 1, // Wake Up
+            0x71: 3, // Alarm/Notification
+            0x9C: 1  // Sensor Alarm
+    ]
+}
+
+def parse(String description) {
+    def result = null
+    if (description.startsWith("Err 106")) {
+        if (state.sec) {
+            log.debug description
+        } else {
+            result = createEvent(
+                    descriptionText: "This sensor failed to complete the network security key exchange. If you are unable to control it via SmartThings, you must remove it from your network and add it again.",
+                    eventType: "ALERT",
+                    name: "secureInclusion",
+                    value: "failed",
+                    isStateChange: true,
+            )
+        }
+    } else if (description != "updated") {
+        def cmd = zwave.parse(description, commandClassVersions)
+        if (cmd) {
+            result = zwaveEvent(cmd)
+        } else {
+            log.debug "No Z-Wave Event for command ${cmd}"
+        }
+    }
+    log.debug "parsed '$description' to $result"
+    return result
+}
+
+def installed() {
+    // Device-Watch simply pings if no device events received for 482min(checkInterval)
+    sendEvent(name: "checkInterval", value: 2 * 4 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+
+    // Request sensor data to initialize the UI
+    def cmds = [
+            zwave.manufacturerSpecificV2.manufacturerSpecificGet(),
+            zwave.sensorBinaryV2.sensorBinaryGet(sensorType: zwave.sensorBinaryV2.SENSOR_TYPE_DOOR_WINDOW),
+            zwave.batteryV1.batteryGet(),
+            zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 1, scale: 1),
+            zwave.wakeUpV1.wakeUpNoMoreInformation()
+    ]
+
+    response(commands(cmds, 1000))
```

Comment:
I'd verify that you can get all of these requests responded to before the device goes to sleep unprompted. If not, it may be wise to do 200ms delays between everything except the wakeUp.

Reply:
If it works as is and we get all the info we need, then don't touch it.

Also you can remove the manufacturer get. This is performed automatically as part of the add process now.

Reply:
`1547e3cb-7337-4cd4-8a43-062d130f182f  11:41:00 AM: debug unsecurely sending WakeUpNoMoreInformation()

1547e3cb-7337-4cd4-8a43-062d130f182f  11:41:00 AM: debug unsecurely sending SensorMultilevelGet(reserved110: 0, reserved212: 0, scale: 1, sensorType: 1)

1547e3cb-7337-4cd4-8a43-062d130f182f  11:41:00 AM: debug unsecurely sending BatteryGet()

1547e3cb-7337-4cd4-8a43-062d130f182f  11:41:00 AM: debug unsecurely sending SensorBinaryGet(sensorType: 10)

1547e3cb-7337-4cd4-8a43-062d130f182f  11:41:00 AM: debug unsecurely sending ManufacturerSpecificGet()`

Above is the log during install. Can I rely on the timestamp? If so, it would seem the delay of 1000 is not being applied.

Reply:
What is the point of the delays? breathing room for the device to respond? I would just remove it if it was me

Reply:
I appear to be getting all of the expected responses with the code as is (with delays). When I tried setting the delay to 0, I missed the last response.

Reply:
I think 200ms is the default, but the `NoMoreInformation` is kind of a special case because a device will often shut down as soon as it sees it, even if it hasn't finished responding to all of its requests.